02-microarray/dimension-reduction_microarray_01_pca.Rmd

```diff
@@ -0,0 +1,334 @@
+
+If you'd like to adapt an example analysis to use a different dataset from [refine.bio](https://www.refine.bio/), we recommend placing the files in the `data/` directory you created and changing the filenames and paths in the notebook to match these files (we've put comments to signify where you would need to change the code).
+We suggest saving plots and results to `plots/` and `results/` directories, respectively, as these are automatically created by the notebook.
+From here you can customize this analysis example to fit your own scientific questions and preferences. 
+
+***
+
+# PCA Visualization - Microarray
+
+## Install libraries
+
+See our Getting Started page with [instructions for package installation](https://alexslemonade.github.io/refinebio-examples/01-getting-started/getting-started.html#what-you-need-to-install) for a list of the other software you will need, as well as more tips and resources.
+
+Attach the packages we need for this analysis:
+
+```{r}
+# Attach the library
+library(ggplot2)
+
+# We will need this so we can use the pipe: %>%
+library(magrittr)
+```
+
+## Import and set up data
+
+Data downloaded from refine.bio include a metadata tab separated values (TSV) file and a data TSV file. 
+This chunk of code will read the both TSV files and add them as data frames to your environment.
+
+```{r}
+# Read in metadata TSV file
+metadata <- readr::read_tsv(file.path(
+  data_dir, # Replace with path to your metadata file
+  "metadata_GSE37382.tsv" # Replace with the name of your metadata file
+))
+
+# Read in data TSV file
+df <- readr::read_tsv(file.path(
+  data_dir, # Replace with path to your data file
+  "GSE37382.tsv" # Replace with the name of your data file
+)) %>%
+  # Tuck away the Gene id column as rownames
+  tibble::column_to_rownames("Gene")
+```
+
+Let's ensure that the metadata and data are in the same sample order. 
+
+```{r}
+# Make the data in the order of the metadata
+df <- df %>%
+  dplyr::select(metadata$geo_accession)
+
+# Check if this is in the same order
+all.equal(colnames(df), metadata$geo_accession)
+```
+
+
+Now we are going to use a combination of functions from base R and the `ggplot2` package to perform and visualize the results of the Principal Component Analysis (PCA) dimension reduction technique on our medulloblastoma samples.
+
+## Perform Principal Components Analysis
+
+In this code chunk, we are going to perform Principal Component Analysis (PCA) on our data and create a data frame using the PCA scores and the variables from our metadata that we are going to use to annotate our plot later.
+We are using the base R `prcomp()` function to perform Principal Component Analysis (PCA) here [@prcomp-function].
+
+```{r}
+# Perform Principal Component Analysis (PCA) using the `prcomp()` function
+pca <- prcomp(
+  t(df), # We have to transpose our data frame to meet the `prcomp()` function requirements
+  scale = TRUE # This tells R that we want the variables scaled to have unit variance
```

Comment:
Was there a particular reason you went with scaled here? I'm inclined to not used scaled since this data is already normalized and should be all on the same scale already. In other words, we shouldn't need `scale = TRUE` because it's normalized, and if the data still looks like it needs scaling, then I think that might indicate other problems with the dataset aka scaling a wonky dataset could hide a red flag. 

There might be a situation to use `scale = TRUE` ? but I'd be inclined to encourage our users to always do `scale = FALSE` first before making that decision.

Reply:
This decision was made based on @jaclyn-taroni's review suggestion [here](https://github.com/AlexsLemonade/refinebio-examples/pull/198#discussion_r479373038), but you do bring up a good point and agree that we shouldn't need it here with the normalized data.

Reply:
Quantile normalization is going to make it such that the samples have the same underlying distributions. The genes won't necessarily end up on the same scale. You could make an argument that it should be "close enough."